### ISSUE TITLE
Trying to delete an item from a flatiter segfaults (migrated from Trac #2045)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3592]
Original ticket http://projects.scipy.org/numpy/ticket/2045
Reported 2012-02-10 by atmention:alex, assigned to unknown.

Tested at the latest master of the github repo:

```
(e268cfc7-ff3f-4d97-a02c-27481c8a8186)alexatmention:alex-gaynor-laptop:~/.virtualenvs/e268cfc7-ff3f-4d97-a02c-27481c8a8186$ python
Python 2.7.2+ (default, Oct  4 2011, 20:06:09) 
[GCC 4.6.1] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> numpy.arange(2)
array([0, 1])
>>> numpy.arange(2).flat
<numpy.flatiter object at 0x2e49910>
>>> del numpy.arange(2).flat[1]
Segmentation fault
```

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2012-02-13

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2012-02-17

Fixed in ae3dd33..b20a013.
